### PR TITLE
Use the diffFormatter hook provided by chronos

### PR DESCRIPTION
**Pull request by markstory**
Hook into chronos' diffFormatter hook. Doing this gives us a pluggable way for to modify timeAgoInWords() and diffForHumans(). Furthermore, it makes diffForHumans() compatible with Cake's i18n libraries.

### CODE REVIEW

File: src/I18n/RelativeTimeFormatter.php [hunk 1]
```diff
@@ -24,6 +25,64 @@
...
     /**
+     * Get the difference in a human readable format.
+     *
+     * @param \Cake\Chronos\ChronosInterface $date The datetime to start with.
+     * @param \Cake\Chronos\ChronosInterface|null $other The datetime to compare against.
+     * @param bool $absolute removes time difference modifiers ago, after, etc
+     * @return string The difference between the two days in a human readable format
+     * @see Cake\Chronos\ChronosInterface::diffForHumans
+     */
+    public function diffForHumans(ChronosInterface $date, ChronosInterface $other = null, $absolute = false)
```
Comment (lorenzo) [inline src/I18n/RelativeTimeFormatter.php]: Should we relax the type hint constraints?

Comment (lorenzo) [inline src/I18n/RelativeTimeFormatter.php]: I'll take the liberty to change that myself :)

File: src/I18n/DateFormatTrait.php [hunk 2]
```diff
@@ -350,6 +350,24 @@ public function jsonSerialize()
     }
 
     /**
+     * Get the difference formatter instance or overwrite the current one.
+     *
+     * @param \Cake\I18n\RelativeTimeFormatter|null $formatter The formatter instance when setting.
+     * @return \Cake\I18n\RelativeTimeFormatter The formatter instance.
+     */
+    public function diffFormatter($formatter = null)
```
Comment (lorenzo) [inline src/I18n/DateFormatTrait.php]: Shouldn't this function be static? It only deals with static properties internally

File: src/I18n/RelativeTimeFormatter.php (hunk 1 above)
Comment (markstory) [inline src/I18n/RelativeTimeFormatter.php]: Yeah I could relax them in Chronos if you think its a good idea.

File: src/I18n/DateFormatTrait.php (hunk 2 above)
Comment (markstory) [inline src/I18n/DateFormatTrait.php]: It's not static in chronos. I can change it there first.

Comment (lorenzo) [inline src/I18n/DateFormatTrait.php]: :+1:

Comment (markstory) [inline src/I18n/DateFormatTrait.php]: Updated now.

**Comment by markstory** [comment]
I think this is the end of the immutable time changes. I will get started on the docs next.

File: src/I18n/RelativeTimeFormatter.php (hunk 1 above)
Comment (lorenzo) [inline src/I18n/RelativeTimeFormatter.php]: Yeah, i do think it is a good idea

**Pull request merged**